homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -554,24 +585,126 @@ async def _update_data(self, item: Pipeline, update_data: dict) -> Pipeline:
 
     def _create_item(self, item_id: str, data: dict) -> Pipeline:
         """Create an item from validated config."""
+        if self._preferred_item is None:
+            self._preferred_item = item_id
         return Pipeline(id=item_id, **data)
 
     def _deserialize_item(self, data: dict) -> Pipeline:
         """Create an item from its serialized representation."""
         return Pipeline(**data)
 
     def _serialize_item(self, item_id: str, item: Pipeline) -> dict:
-        """Return the serialized representation of an item."""
+        """Return the serialized representation of an item for storing."""
         return item.to_json()
 
+    async def async_delete_item(self, item_id: str) -> None:
+        """Delete item."""
+        if self._preferred_item == item_id:
+            raise PipelinePreferred(item_id)
+        await super().async_delete_item(item_id)
+
+    @property
+    @callback
+    def preferred_item(self) -> str | None:
+        """Get the id of the preferred item."""
+        return self._preferred_item
+
+    @preferred_item.setter
+    @callback
+    def preferred_item(self, item_id: str) -> None:
```

Comment:
Can we make this a method. I don't like side effects from setting a property.

Reply:
Sure, changed to a normal function